### PR TITLE
feat(types,clerk-js): Introduce OrganizationSuggestion resource and types

### PR DESCRIPTION
**Pull request by chanioxaris**
## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [x] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

We introduce a new OrganizationSuggestion resource and the corresponding types, in order to preview user with suggestions of organizations they can join. The available methods for the time being are retrieve() which return a list of organization suggestions of the user and accept() which allows a user to request to join the organization

### CODE REVIEW

File: packages/types/src/json.ts [hunk 1]
```diff
@@ -363,6 +364,21 @@ export interface OrganizationDomainJSON extends ClerkResourceJSON {
   updated_at: number;
 }
 
+export interface OrganizationSuggestionJSON extends ClerkResourceJSON {
+  object: 'organization_suggestion';
+  id: string;
+  public_organization_data: {
```
Comment (panteliselef) [inline packages/types/src/json.ts:370]: Shall we create a type for this ? I think with this PR we will be using the same code in about 4 places

File: packages/types/src/organizationSuggestion.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import type { ClerkResource } from './resource';
+
+export type OrganizationSuggestionStatus = 'pending' | 'accepted';
```
Comment (panteliselef) [inline packages/types/src/organizationSuggestion.ts:3]: Will this in the feature include 'revoked' or 'rejected' ?

I am trying to figure out why we cannot reuse `OrganizationInvitationStatus`

**Review by panteliselef — APPROVED** [review]
Looks good 💯, added some comments asking for clarifications

File: packages/types/src/organizationSuggestion.ts (hunk 2 above)
Comment (chanioxaris) [reply packages/types/src/organizationSuggestion.ts:3]: Yes we are planning to introduce more statuses in the future, so it's better to create an explicit type instead

File: packages/shared/src/hooks/useOrganizationList.tsx [hunk 3]
```diff
@@ -64,22 +82,31 @@ export const useOrganizationList: UseOrganizationList = params => {
...
+    isFetching: isomorphicIsFetchingInvitations,
+    isError: isomorphicIsErrorInvitations,
+    page: isomorphicPageInvitations,
+    pageCount: pageCountInvitations,
+    fetchPage: isomorphicSetPageInvitations,
+    fetchNext: fetchNextInvitations,
+    fetchPrevious: fetchPreviousInvitations,
+    hasNextPage: hasNextPageInvitations,
+    hasPreviousPage: hasPreviousPageInvitations,
+    unstable__mutate: unstableMutateInvitations,
```
Comment (panteliselef) [inline packages/shared/src/hooks/useOrganizationList.tsx:109]:
```suggestion
userInvitations
```

**Review by panteliselef — APPROVED** [review]

File: packages/types/src/json.ts (hunk 1 above)
Comment (chanioxaris) [reply packages/types/src/json.ts:370]: Sure let me do it in a following refactor PR to unblock yours for now

**Review by SokratisVidros — APPROVED** [review]

**Review by panteliselef — APPROVED** [review]

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**